k2/csrc/context.cuh

```diff
@@ -129,6 +131,86 @@ class Context : public std::enable_shared_from_this<Context> {
   virtual void Sync() const {}
 };
 
+// TODO(haowen): most of implementations below should be updated later.
+class CpuContext : public Context {
+ public:
+  ContextPtr GetCpuContext() override { return nullptr; }
+  ContextPtr GetPinnedContext() override { return nullptr; }
+  DeviceType GetDeviceType() const override { return kCpu; }
+
+  void *Allocate(std::size_t bytes, void **deleter_context) override {
+    void *p = nullptr;
+    if (bytes) {
+      int32_t ret = posix_memalign(&p, kAlignment, bytes);
+      K2_DCHECK_EQ(ret, 0);
+    }
+    if (deleter_context) *deleter_context = nullptr;
+    return p;
+  }
+
+  bool IsCompatible(const Context &other) const override {
+    return other.GetDeviceType() == kCpu;
+  }
+
+  void Deallocate(void *data, void * /*deleter_context*/) override {
+    free(data);
+  }
+};
+
+#ifdef __CUDACC__
```

Comment:
Do we really need this? I just thought we always compile code with nvcc..

Reply:
This should not be here.  meixu, you shouldn't be attempting changes like this without asking, anyway.

Reply:
Oh, no wonder you guys keep ignoring gcc.. Deleted here and others.